vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -221,6 +230,68 @@ public String getPath() {
         return getElement().getAttribute("path");
     }
 
+    private void doSetPath(String path) {
+        if (path == null) {
+            getElement().removeAttribute("path");
+        } else {
+            getElement().setAttribute("path",
+                    sanitizePath(refreshPathParameters(path)));
+        }
+    }
+
+    /**
+     * Gets the {@link QueryParameters} of this item.
+     *
+     * @return an optional of {@link QueryParameters}, or an empty optional if
+     *         there are no query parameters set
+     * @see #setQueryParameters(QueryParameters)
+     */
+    public Optional<QueryParameters> getQueryParameters() {
+        return Optional.ofNullable(queryParameters);
+    }
+
+    /**
+     * Sets the {@link QueryParameters} of this item.
+     * <p>
+     * The query string will be generated from
+     * {@link QueryParameters#getQueryString()} and will be appended to the
+     * {@code path} attribute of this item.
+     *
+     * @param queryParameters
+     *            the query parameters object, or {@code null} to remove
+     *            existing query parameters
+     */
+    public void setQueryParameters(QueryParameters queryParameters) {
+        this.queryParameters = queryParameters;
+        refresh();
+    }
+
+    private void refresh() {
+        if (view == null) {
+            setPath(customPath);
+        } else {
+            setPath(view);
+        }
+    }
+
+    private String sanitizePath(String path) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        return UrlUtil.encodeURI(path);
+    }
+
+    private String refreshPathParameters(String path) {
```

Comment:
As far as I understand, query parameters can be also inside path parts, e.g.: `/users/:id/edit`

Reply:
This makes me think that the only way to set path aliases should be through `setPath` which would accept query parameters as the second argument. In my opinion, it doesn't make practical sense to allow setting aliases from different views separately from the path view.

```diff
- setPathAliases(Class<? extends Component> views...)
+ setPath(Class<? extends Component> view)
+ setPath(Class<? extends Component> view, QueryParameters queryParameters)
```

Reply:
Makes sense. I will apply this suggestion to the path alias PR after this one is merged.

> This makes me think that the only way to set path aliases from a view should be through `setPath`, which would also accept query parameters as the second argument. In my opinion, it doesn't seem practical to allow setting aliases from different views separately from the view that defines the main path.